theano/tensor/nnet/neighbours.py

```diff
@@ -111,6 +111,43 @@ def grad(self, inp, grads):
                 return [neibs2images(gz, neib_shape, x.shape, mode=self.mode),
                         grad_undefined(self, 1, neib_shape),
                         grad_undefined(self, 2, neib_step)]
+
+        if self.mode in ['valid']:
+            # Iterate over neighborhood positions, summing contributions.
+            def pos2map(pidx, pgz, prior_result, neib_shape, neib_step):
+                '''
+                Helper function that adds gradient contribution from a single
+                neighborhood position i,j.
+                pidx = Index of position within neighborhood.
+                pgz  = Gradient of shape (1, batch_size*num_channels*neibs)
```

Comment:
currently, the pgz shape is a vector of lenght (batch_size*num_channels*neibs) (drop the 1)